libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/GutenbergEditorFragment.java

```diff
@@ -509,6 +509,7 @@ public void onClick(DialogInterface dialog, int id) {
             public void onClick(DialogInterface dialog, int id) {
                 dialog.dismiss();
                 mEditorFragmentListener.onMediaDeleted(String.valueOf(mediaId));
+                mFailedMediaIds.remove(String.valueOf(mediaId));
```

Comment:
Note (unrelated) for clarity: while testing this PR step by step, I noticed we should also be doing this when the user taps on the trash bin icon to delete the block (as opposed to "removing" the image from the block, without removing the actual empty image block). 
Given that action isn't related to what this PR specifically fixes, going to check and open a new issue separately if merits.

Reply:
Tracked this in https://github.com/wordpress-mobile/gutenberg-mobile/issues/1595